### PR TITLE
Move storiface into Curio

### PR DESCRIPTION
**Pull request by magik6k**
This PR moves storiface (and legacy partialfile) from lotus into curio to untie it from lotus and get us all the flexibility needed with storage types.

Needed for unseal

### CODE REVIEW

File: lib/storiface/filetype.go [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+// FTUnsealed represents an unsealed sector file.
+// FTSealed represents a sealed sector file.
+// FTCache represents a cache sector file.
+// FTUpdate represents an update sector file.
```
Comment (snadrus) [inline lib/storiface/filetype.go:14]: what's an update?

File: lib/storiface/storage.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+type StorageConfig struct {
+	StoragePaths []LocalPath
+}
+
+type LocalPath struct {
+	Path string
+}
+
+// LocalStorageMeta [path]/sectorstore.json
+type LocalStorageMeta struct {
```
Comment (snadrus) [inline lib/storiface/storage.go:193]: Is this a subset of StorageInfo ?
It may be nice to avoid dup by including the simpler type in the advanced type,.

**Review by snadrus — APPROVED** [review]

File: lib/storiface/filetype.go (hunk 1 above)
Comment (LexLuthr) [reply lib/storiface/filetype.go:14]: Snap sealed file

File: lib/storiface/storage.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+type ReplicaUpdateProof []byte
+type ReplicaVanillaProofs [][]byte
+
+type ReplicaUpdateOut struct {
+	NewSealed   cid.Cid
+	NewUnsealed cid.Cid
+}
+
+type Sealer interface {
```
Comment (LexLuthr) [inline lib/storiface/storage.go:76]: Do we need this interface anymore?

File: tasks/window/fault_tracker.go
Comment (LexLuthr) [inline tasks/window/fault_tracker.go:1]: Why do we need this?

**Review by LexLuthr — APPROVED** [review]
Looks good. Maybe some cleanup of things not used by Curio?

File: lib/storiface/filetype.go (hunk 1 above)
Comment (magik6k) [reply lib/storiface/filetype.go:14]: Updated replica (Snap sealed file)

File: lib/storiface/storage.go (hunk 2 above)
Comment (magik6k) [reply lib/storiface/storage.go:193]: This one in the on-disk path config, the second one contains runtime information about the path

File: lib/storiface/storage.go (hunk 3 above)
Comment (magik6k) [reply lib/storiface/storage.go]: Oh we don't

**Pull request merged**